iroh-bytes/Cargo.toml

```diff
@@ -13,8 +13,7 @@ rust-version = "1.66"
 
 [dependencies]
 anyhow = { version = "1", features = ["backtrace"] }
-bao-tree = { version = "0.5.0", features = ["tokio_fsm"], default-features = false }
-blake3 = "1.3.3"
```

Comment:
I believe there are a couple of other users (`iroh` crate at least) that use `blake3` directly, should be unified, to avoid duplicate dependencies

Reply:
Not sure what to do about this. The way it is now,

- bao-tree uses iroh-blake3 instead of blake3
- bao-tree reexports this as blake3
- both iroh-bytes and iroh use bao-tree and the reexport of blake3

If some other crate uses both iroh and upstream blake3, it will work but you will have a duplicate dependency. But what can we do about it? Encourage people to use our fork or reexport blake3 in iroh somewhere?

Reply:
It is fine if folks depending on iroh endup with duplicate deps, but at least in our tree it should be clean

Reply:
There is the case where people might want to convert from a blake3::Hash to an iroh_bytes::Hash. That won't work unless they are using iroh-blake3. But this should be rare, so maybe fix it only if there is no chance of upstreaming the blake3 changes.

The blake3 changes are *really* minimal, so I hope we can get them in.